### PR TITLE
Stub out basic docs structure

### LINKED ISSUE #30: Define and create the documentation structure
The current documentation structure is the default one sphinx creates when setting up a new project. Before we start diving into adding documentation about each command and its sub-commands, etc. We need to come up with the structure on where content will be placed.

This issue is to create the structure for the documentation.

Possible sections:
- Introduction
- Installation
- Quickstart
- Commands

### PR DESCRIPTION
**Pull request by tehsmyers**
This lays down a basic docs structure as outlined in
https://github.com/rhpit/manageiq-cli/issues/30. I've made some
arbitrary choices that are up for discussion here, including:

- Using sphinx-apidoc to delegate api documentation to sphinx's own
  :auto*: directives
- Avoiding documenting the actual CLI commands, because that looks a
  little difficult to do in a way that's easy to maintain -- I punted
  that to basically say "run miqcli --help".
- Using pbr.packaging's "get_version" function to define the VERSION
  constant, as well as set the version in the docs config
- Documenting the constants in general with the "#:" syntax
- Other stuff caught in review that I'm forgetting :)

Closes #30

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+sphinx
```
Comment (ryankwilliams) [inline doc/requirements.txt]: We should not need to define sphinx within the requirements.txt. As far as I know any package defined within requirements.txt means they are needed to run the application. Sphinx is just needed for development when we need to build & release.

The sphinx package is already included in the test-requirements.txt. Which contains other packages for development.

```
coverage!=4.4,>=4.0
mock>=2.0.0
sphinx>=1.6.2
testtools>=2.2.0
flake8>=3.5.0
```
What do you think @seandst?

**Review by ryankwilliams — CHANGES_REQUESTED** [review]

File: doc/requirements.txt (hunk 1 above)
Comment (tehsmyers) [reply doc/requirements.txt]: Those are specifically docs requirements, which I added before creating a docs dependency on miqcli itself. At that point they became redundant, so the file should just go away. +1

File: miqcli/constants.py [hunk 2]
```diff
@@ -13,22 +13,46 @@
...
+miqcli.constants values will differ from installation to installation.
+"""
 import os
 
-VERSION = '0.0.0'
+from pbr.packaging import get_version
+
+
+#: Installed version of miqcli
+VERSION = get_version('miqcli')
```
Comment (tehsmyers) [inline miqcli/constants.py]: I just noticed that if we do this, we should swap VERSION and PACKAGE, so VERSION is `get_version(PACKAGE)`

**Review by tvieira — APPROVED** [review]
LGTM

**Comment by ryankwilliams** [comment]
LGTM

**Review by ryankwilliams — APPROVED** [review]

**Comment by tvieira** [comment]
Hey @seandst , I think one of the latest commits changed the `constants.py`, very small change. Feel free to rebase with that and do a "squash and merge".

**Pull request merged**